whipper/command/mblookup.py

```diff
@@ -12,35 +12,52 @@ class MBLookup(BaseCommand):
 
     def add_arguments(self):
         self.parser.add_argument(
-            'mbdiscid', action='store', help="MB disc id to look up"
+            'id', action='store', help="MB disc id or release id to look up"
         )
 
+    def _printMetadata(self, md):
+        """
+        Print out metadata received in a sensible way
+
+        :param md: MusicBrainz's metadata about the disc
+        :type md: `DiscMetadata`
+        rtype: None
+        """
+        print('    Artist: %s' % md.artist.encode('utf-8'))
+        print('    Title:  %s' % md.title.encode('utf-8'))
+        print('    Type:   %s' % str(md.releaseType).encode('utf-8'))
+        print('    URL:    %s' % md.url)
+        print('    Tracks: %d' % len(md.tracks))
+        if md.catalogNumber:
+            print('    Cat no: %s' % md.catalogNumber)
+        if md.barcode:
+            print('    Barcode: %s' % md.barcode)
+
+            for j, track in enumerate(md.tracks):
+                print('      Track %2d: %s - %s' % (
+                    j + 1, track.artist.encode('utf-8'),
+                    track.title.encode('utf-8')
+                ))
+        return None
+
     def do(self):
         try:
-            discId = str(self.options.mbdiscid)
+            id = str(self.options.id)
         except IndexError:
-            print('Please specify a MusicBrainz disc id.')
+            print('Please specify a MusicBrainz disc id or release id.')
             return 3
 
-        metadatas = musicbrainz(discId)
-
-        print('%d releases' % len(metadatas))
-        for i, md in enumerate(metadatas):
-            print('- Release %d:' % (i + 1, ))
-            print('    Artist: %s' % md.artist.encode('utf-8'))
-            print('    Title:  %s' % md.title.encode('utf-8'))
-            print('    Type:   %s' % str(md.releaseType).encode('utf-8'))  # noqa: E501
-            print('    URL:    %s' % md.url)
-            print('    Tracks: %d' % len(md.tracks))
-            if md.catalogNumber:
-                print('    Cat no: %s' % md.catalogNumber)
-            if md.barcode:
-                print('    Barcode: %s' % md.barcode)
-
-                for j, track in enumerate(md.tracks):
-                    print('      Track %2d: %s - %s' % (
-                        j + 1, track.artist.encode('utf-8'),
-                        track.title.encode('utf-8')
-                    ))
-
+        # release id is is 36 characters long and disc id is 28 characters long
+        # see https://musicbrainz.org/doc/MusicBrainz_Identifier
+        if len(id) == 36:
+            md = getReleaseMetadata(id)
+            if md:
+                self._printMetadata(md)
+        elif len(id) == 28:
```

Comment:
Maybe you can use a RegEX instead of a length check for the Disc ID (I'm not sure this RegEX is correct):

```regex
[\dA-Za-z._]{27}-
```

Reply:
I think the disc id doesn't always end with '-'

Reply:
After more consideration I think the RegEX posted above is correct.
The character set is different from the base64 standard as detailed here: [Disc ID Calculation](https://musicbrainz.org/doc/Disc_ID_Calculation)
> **Step 2: Base64-encoding of the hash**
> 
> The resulting 20 byte SHA-1 signature is converted into a Base64 encoded string of printable ASCII characters, which is the disc ID. All disc ID strings are thus exactly 28 characters long. In the above audio CD example the disc ID is `49HHV7Eb8UKF3aQiNmu1GR8vKTY-`. For details about Base64, please see [RFC 4648](https://tools.ietf.org/html/rfc4648) or the reasonable good [Wikipedia article](https://en.wikipedia.org/wiki/Base64).
> 
> **Important note:** The Base64 encoding used by MusicBrainz is not the same one as specified in [RFC 4648](https://tools.ietf.org/html/rfc4648). The specification uses `+`, `/`, and `=` characters, all of which are special HTTP/URL characters. To avoid the problems with dealing with that, MusicBrainz uses `.`, `_`, and `-` instead. For details on this, please refer to [base64.c](https://github.com/metabrainz/libdiscid/blob/master/src/base64.c) in the [libdiscid source code](https://github.com/metabrainz/libdiscid).

Reply:
Comment updated.

Reply:
Okay i fixed it in a new commit